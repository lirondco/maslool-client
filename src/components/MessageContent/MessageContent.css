.MessageContent {
    margin: 2% 20% 2% 20%;
    border: black solid 2px;
    border-radius: 20px;
    padding: 3%;
}

.MessageContent p, .MessageContent span, .MessageContent a {
    background-color: white;
}

.MessageContent p {
    margin: 2%;
    padding: 1%;
    border: black solid 1px;
    border-radius: 20px;
}

.MessageContent a {
    color: blue;
    text-decoration: underline;
}

.MessageContent a:hover {
    color: purple;
}

.MessageContent span {
    font-weight: bold;
}