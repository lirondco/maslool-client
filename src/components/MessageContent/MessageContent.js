import React, { Component } from "react";
import "./MessageContent.css";

export default class MessageContent extends Component {
  static defaultProps = {
    onDeleteClick: () => {},
  };

  render() {
    const pending = this.props.pending;

    return (
      <div className="MessageContent">
        <p>
          {" "}
          From: <span> {pending.user.username} </span>
        </p>
      </div>
    );
  }
}
