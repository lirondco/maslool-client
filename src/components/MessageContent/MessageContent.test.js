import React from "react";
import { shallow } from "enzyme";
import toJson from "enzyme-to-json";
import MessageContent from "./MessageContent";

describe("MessageContent component", () => {
  const props = {
    id: 9,
    message: "This is a message",
    user: {
      id: 1,
      username: "test_user",
      email: "test@user.no",
      join_date: "2021-03-11T06:47:06.477Z",
      admin: false,
      banned: false,
    },
  };

  it("renders the message given props", () => {
    const wrapper = shallow(<MessageContent pending={props} />)
    expect(toJson(wrapper)).toMatchSnapshot()
  });

  it('button works when clicked', () => {
      const wrapper = shallow(<MessageContent pending={props} />)
      window.confirm = jest.fn(() => true)
      wrapper.find('ForwardRef').simulate('click', { preventDefault() {}})
      expect(window.confirm).toBeCalled()
  })
});
