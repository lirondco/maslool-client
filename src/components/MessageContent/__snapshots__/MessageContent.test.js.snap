// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MessageContent component renders the message given props 1`] = `
<div
  className="MessageContent"
>
  <p>
    From: 
    <span>
      test_user
    </span>
  </p>
  <p>
    Message user: 
    <span>
      <a
        href="mailto: test@user.no"
      >
        test@user.no
      </a>
    </span>
  </p>
  <p
    style={
      Object {
        "whiteSpace": "pre-wrap",
      }
    }
  >
    This is a message
  </p>
  <ForwardRef
    onClick={[Function]}
    type="button"
  >
    Delete Message
  </ForwardRef>
</div>
`;
